L1Trigger/L1TZDC/plugins/L1TZDCAnalyzer.cc

```diff
@@ -0,0 +1,200 @@
+  void L1TZDCAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+    using namespace edm;
+
+    std::stringstream text;
+
+    //    Handle<EtSumBxCollection> sums;
+    Handle<BXVector<l1t::EtSum> > sums;
+    iEvent.getByToken(m_sumToken, sums);
+    //    Handle<BXVector<l1t::EtSum> > sumsM;
+    //    iEvent.getByToken(m_sumMToken, sumsM);
+
+    for (int ibx = sums->getFirstBX(); ibx <= sums->getLastBX(); ++ibx) {
+      for (auto itr = sums->begin(ibx); itr != sums->end(ibx); ++itr) {
+        if (itr->getType() == l1t::EtSum::EtSumType::kZDCP)
+          m_zdcEtSumP[ibx] = itr->hwPt();
+        if (itr->getType() == l1t::EtSum::EtSumType::kZDCM)
+          m_zdcEtSumM[ibx] = itr->hwPt();
+      }
+    }
+    /*
+    for (int ibx = sumsM->getFirstBX(); ibx <= sumsM->getLastBX(); ++ibx) {
+      int sumCounterM = 0;
+      for (auto itr = sumsM->begin(ibx); itr != sumsM->end(ibx); ++itr) {
+	m_zdcEtSumM[ibx] = itr->hwPt();
+	++sumCounterM;
+      }
+    }
+    */
+    m_zdcEtSumTree_p->Fill();
+
+    if (doText_)
+      edm::LogVerbatim("L1TCaloEvents") << text.str();
+  }
+
+  // ------------ method called once each job just before starting event loop  ------------
+  void L1TZDCAnalyzer::beginJob() {
+    m_zdcEtSumTree_p = m_fs->make<TTree>("zdcEtSumTree", "");
+    m_zdcEtSumTree_p->Branch("zdcEtSumP", m_zdcEtSumP, ("zdcEtSumP[" + std::to_string(m_maxBPX) + "]/F").c_str());
+    m_zdcEtSumTree_p->Branch("zdcEtSumM", m_zdcEtSumM, ("zdcEtSumM[" + std::to_string(m_maxBPX) + "]/F").c_str());
+  }
+
+  // ------------ method called once each job just after ending the event loop  ------------
+  void L1TZDCAnalyzer::endJob() {}
+
+  // ------------ method called when starting to processes a run  ------------
+  /*
+    void 
+    L1TZDCAnalyzer::beginRun(edm::Run const&, edm::EventSetup const&)
+    {
+    }
+  */
+
+  // ------------ method called when ending the processing of a run  ------------
+  /*
+    void 
+    L1TZDCAnalyzer::endRun(edm::Run const&, edm::EventSetup const&)
+    {
+    }
+  */
+
+  // ------------ method called when starting to processes a luminosity block  ------------
+  /*
+    void 
+    L1TZDCAnalyzer::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+    {
+    }
+  */
+
+  // ------------ method called when ending the processing of a luminosity block  ------------
+  /*
+    void 
+    L1TZDCAnalyzer::endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+    {
+    }
+  */
+
+  // ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+  void L1TZDCAnalyzer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    //The following says we do not know what parameters are allowed so do no validation
+    // Please change this to state exactly what you do use, even if it is no parameters
+    edm::ParameterSetDescription desc;
+    desc.setUnknown();
+    descriptions.addDefault(desc);
```

Comment:
please implement a real `fillDescriptions()` method

Reply:
Should be fixed in latest commit